File: plugins/dotnet/robocode.dotnet.napi/src/net/sf/robocode/security/IHiddenSpecificationHelper.cs
using robocode.control;

namespace net.sf.robocode.security
{
    public interface IHiddenSpecificationHelper
    {
        RobotSpecification createSpecification(object obj, string str1, string str2, string str3, string str4,
                                               string str5, string str6, string str7, string str8);

        object getFileSpecification(RobotSpecification rs);
        string getTeamName(RobotSpecification rs);
        void setTeamName(RobotSpecification rs, string str);
    }
}

File: robocode.api/src/main/java/net/sf/robocode/security/IHiddenSpecificationHelper.java
/**
 * Copyright (c) 2001-2020 Mathew A. Nelson and Robocode contributors
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * https://robocode.sourceforge.io/license/epl-v10.html
 */
package net.sf.robocode.security;


import robocode.control.RobotSpecification;


/**
 * @author Pavel Savara (original)
 */
public interface IHiddenSpecificationHelper {

	RobotSpecification createSpecification(Object specification, String name, String author, String webpage, String version, String robocodeVersion, String jarFile, String fullClassName, String description);
	Object getFileSpecification(RobotSpecification specification);
	void setTeamName(RobotSpecification specification, String teamName);
	String getTeamName(RobotSpecification specification);
}
